homeassistant/core.py

```diff
@@ -1207,51 +1208,25 @@ def _async_log_running_tasks(self, stage: str) -> None:
             _LOGGER.warning("Shutdown stage '%s': still running: %s", stage, task)
 
 
+@dataclasses.dataclass(frozen=True, kw_only=True)
```

Comment:
`kw_only` is a breaking change

Reply:
https://docs.python.org/3/library/dataclasses.html#frozen-instances

> There is a tiny performance penalty when using frozen=True: [__init__()](https://docs.python.org/3/reference/datamodel.html#object.__init__) cannot use simple assignment to initialize fields, and must use __setattr__().

This is a performance sensitive code path. Have you tested the performance of this change?